exasol_integration_test_docker_environment/test/test_hash_temp_dir.py

```diff
@@ -206,12 +216,161 @@ def test_directory_file_names_not_equal_to_dir_names(self):
+            f.write("test")
+            hash1_content_only = hasher_content_only.hash([PathMapping("test.txt", test_file)])
+        os.chdir(self.test_dir2)
+
+        with open(test_file, "w") as f:
+            f.write("test")
+            hash2_content_only = hasher_content_only.hash([PathMapping("test.txt", test_file)])
+        os.chdir(old_pwd)
+        ascii_hash1_content_only = base64.b32encode(hash1_content_only).decode("ASCII")
+        ascii_hash2_content_only = base64.b32encode(hash2_content_only).decode("ASCII")
+        self.assertEqual(ascii_hash1_content_only, ascii_hash2_content_only)
+
+    def test_duplicated_file_mapping_raises_exception(self):
+        """
+        Test that a duplicated mapping raises an exception.
+        """
+        hasher_content_only = \
+            FileDirectoryListHasher(followlinks=True,
+                                    hashfunc="sha256",
+                                    hash_file_names=True,
+                                    hash_directory_names=True,
+                                    hash_permissions=True,
+                                    use_relative_paths=True)
+        test_file1 = f"{self.test_dir1}/level0/level1_0/test.txt"
+        with open(test_file1, "w") as f:
+            f.write("test")
+        test_file2 = f"{self.test_dir1}/level0/level1_1/test.txt"
+        with open(test_file2, "w") as f:
+            f.write("test")
+
+        self.assertRaises(AssertionError, lambda: hasher_content_only.hash([PathMapping("test.txt", test_file1),
+                                                                            PathMapping("test.txt", test_file2)]))
+
+    def test_duplicated_path_mapping_raises_exception(self):
+        """
+        Test that a duplicated mapping raises an exception.
+        """
+        hasher_content_only = \
+            FileDirectoryListHasher(followlinks=True,
+                                    hashfunc="sha256",
+                                    hash_file_names=True,
+                                    hash_directory_names=True,
+                                    hash_permissions=True,
+                                    use_relative_paths=True)
+        test_file1 = f"{self.test_dir1}/level0/level1_0/test/test.txt"
+        p = Path(f"{self.test_dir1}/level0/level1_0/test")
+        p.mkdir()
+        with open(test_file1, "w") as f:
+            f.write("test")
+        test_file2 = f"{self.test_dir1}/level0/level1_1/test/test.txt"
+        p = Path(f"{self.test_dir1}/level0/level1_1/test")
+        p.mkdir()
+        with open(test_file2, "w") as f:
+            f.write("test")
+
+        path1 = f"{self.test_dir1}/level0/level1_0/"
+        path2 = f"{self.test_dir1}/level0/level1_1/"
+        self.assertRaises(AssertionError, lambda: hasher_content_only.hash([PathMapping("test", path1),
+                                                                            PathMapping("test", path2)]))
```

Comment:
You should also add a test for duplicated path mapping with subpaths as key

Reply:
done